src/Quasar/Mount/SparkFTP/Gen.purs

```diff
@@ -0,0 +1,38 @@
+{-
+Copyright 2017 SlamData, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-}
+
+module Quasar.Mount.SparkFTP.Gen where
+
+import Prelude
+
+import Control.Monad.Gen (class MonadGen)
+import Control.Monad.Gen as Gen
+import Control.Monad.Gen.Common as GenC
+import Control.Monad.Rec.Class (class MonadRec)
+import Data.StrMap.Gen as SMG
+import Quasar.Mount.Common.Gen (genAlphaNumericString, genHost, genAnyPath)
+import Quasar.Mount.SparkFTP as SFTP
+
+genConfig ∷ ∀ m. MonadGen m ⇒ MonadRec m ⇒ m SFTP.Config
+genConfig = do
+  sparkHost ← genHost
+  ftpHost ← genHost
+  path ← genAnyPath
+  withCreds ← Gen.chooseBool
+  props ← SMG.genStrMap genAlphaNumericString (GenC.genMaybe genAlphaNumericString)
+  user ← genAlphaNumericString
+  password ← genAlphaNumericString
+  pure { sparkHost, ftpHost, path, user, password, props }
```

Comment:
This could be written as 

``` purescript
{ sparkHost: _, ftpHost: _, path: _, user: _, password: _, props: _ }
  <$> genHost
  <*> genHost
  <*> genAnyPath
  <*> genAlphaNumericString
  <*> genAlphaNumericString
  <*> SMG.genStrMap genAlphaNumericString (GenC.genMaybe genAlphaNumericString)
```

Or at least we could remove the `withCreds ← Gen.chooseBool` line since it's not used now :)